python/cuml/linear_model/linear_regression_mg.pyx

```diff
@@ -313,7 +316,7 @@ class LinearRegressionMG:
 
         X_ptr = X.ctypes.data
         y_ptr = y.ctypes.data
-        gpu_ids_ptr = np.array(gpu_ids).ctypes.data
+        gpu_ids_ptr = np.array(gpu_ids, dtype=np.int32).ctypes.data
```

Comment:
```suggestion
        gpu_id_32 = np.array(gpu_ids, dtype=np.int32)
        gpu_ids_ptr = gpu_id_32.ctypes.data
```

Otherwise `np.array(gpu_ids, dtype=np.int32)` could get freed causing issues.